ydb/docs/ru/core/yql/reference/yql-core/udf/list/knn.md

```diff
@@ -144,14 +147,35 @@ $vector = [1.f, 2.f, 3.f, 4.f];
 UPSERT INTO Facts (id, user, fact, embedding) 
 VALUES (123, "Williams", "Full name is John Williams", Untag(Knn::ToBinaryStringFloat($vector), "FloatVector"));
 ```
+{% else %}
+### Декларация данных
+
+```sql
+$vector = [1.f, 2.f, 3.f, 4.f];
+$facts = AsList(
+    AsStruct(
+        123 AS id,  -- Id of fact
+        "Williams" AS user,  -- User name
+        "Full name is John Williams" AS fact,  -- Human-readable description of a user fact
+        Knn::ToBinaryStringFloat($vector) AS embedding,  -- Binary representation of embedding vector
+    ),
+);
+```
+{% endif %}
 
 ### Точный поиск K ближайших векторов
 
 ```sql
 $K = 10;
 $TargetEmbedding = Knn::ToBinaryStringFloat([1.2f, 2.3f, 3.4f, 4.5f]);
 
-SELECT * FROM Facts
+SELECT *
+{% if backend_name == "YDB" %}
```

Comment:
These if's are not rendering as intended: https://ydb-platform--ydb.viewer.diplodoc.com/en/yql/reference/udf/list/knn?revision=pr-8425-e351304a25dc532e9b60cdcd72ef625ebe0c56b8#exact-search-of-k-nearest-vectors

Reply:
Fixed